### ISSUE TITLE
Problem: Job Scheduling via Cron Notation complicated

### ISSUE THREAD

**Issue by maennchen** [quantum-elixir/quantum-core#268]
## Description

Quantum currently allows configuration via Cron Notation. This works very well for statically recurring events, but doesn't work very well for dynamic ones.

If you for example want to run a dynamic job once every day for 7 days until it stops, that's very complicated to do.

## Implementation

Generic way of configuring such jobs should be provided. Any Ideas how to do that?

## PR

We'd be happy to receive a PR (after the implementation details are cleared). If this is a popular request please leave a 👍 so that we see that this is important to our users and implement it ourselves.

**Comment by pyatkov** [comment]
I strongly agree this is important functionality that I'm missing. I have some ideas on how this could be implemented. For very simplistic tasks (like no exeution guarantees) I wrote a home-grown library (that is nowhere near community standards, but allows me to do my stuff) that kind of tracks execution period and transitions to inactive state when the period expires. Probably, will look into it when other activities are drained.

**Comment by maennchen** [comment]
I would like to implement a generic scheduling interface for `v3.0.0`, that could be extended with other providers in the future.
